app/presenters/user_profile_presenter.rb

```diff
@@ -0,0 +1,46 @@
+class UserProfilePresenter
+  def initialize(current_user, user, view_context)
+    @current_user = current_user
+    @user = user
+    @v = view_context
+  end
+
+  def display_profile_options
+    unless @current_user
```

Comment:
just to cut down on the indenting and `end`s what do you think about `return unless @current_user`?